nsxt/data_source_nsxt_policy_tier0_gateway_interface.go

```diff
@@ -0,0 +1,130 @@
+	// t0interface "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+)
+
+func dataSourceNsxtPolicyTier0GatewayInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNsxtPolicyTier0GatewayInterfaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"id":           getDataSourceIDSchema(),
+			"display_name": getDataSourceDisplayNameSchema(),
+			"description":  getDataSourceDescriptionSchema(),
+			"t0_gateway_name": {
+				Type:        schema.TypeString,
+				Description: "The name of the Tier0 gateway where the interface is linked",
+				Required:    true,
+			},
+			"path": getPathSchema(),
+			"edge_cluster_path": {
+				Type:        schema.TypeString,
+				Description: "The path of the edge cluster connected to the Tier0 gateway linked to this interface",
+				Optional:    true,
+				Computed:    true,
+			},
+			"segment_path": {
+				Type:        schema.TypeString,
+				Description: "The path of the edge cluster connected to the Tier0 gateway linked to this interface",
+				Optional:    true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourceNsxtPolicyTier0GatewayInterfaceRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+	converter := bindings.NewTypeConverter()
+	interfaceName := d.Get("display_name").(string)
+	interfaceId := d.Get("id").(string)
+	t0Gw := d.Get("t0_gateway_name").(string)
+
+	// Get the T0 gateway ID as in some case the gw name and the id are different
+	t0GwObj := GenericResourceData{
+		DisplayName: t0Gw,
+		Id:          "",
+	}
+	gwObjList, err := t0GwObj.policyGenericDataSourceResourceRead(connector, getSessionContext(d, m), "Tier0", nil)
+	if err != nil {
+		return err
+	}
+	t0GwList := []model.Tier0{}
+	for _, obj := range gwObjList {
+		dataValue, errors := converter.ConvertToGolang(obj, model.Tier0BindingType())
+		if len(errors) > 0 {
+			return errors[0]
+		}
+		curGw := dataValue.(model.Tier0)
+		if *curGw.DisplayName == t0Gw {
+			t0GwList = append(t0GwList, dataValue.(model.Tier0))
+		}
+	}
+
+	// Get the interface
+	interfaceObj := GenericResourceData{
+		DisplayName: interfaceName,
+		Id:          interfaceId,
+	}
+	interfaceObjList, err := interfaceObj.policyGenericDataSourceResourceRead(connector, getSessionContext(d, m), "Tier0Interface", nil)
+	if err != nil {
+		return err
+	}
+
+	isOp := false
+	for _, obj := range interfaceObjList {
+		dataValue, errors := converter.ConvertToGolang(obj, model.Tier0InterfaceBindingType())
+		if len(errors) > 0 {
+			return errors[0]
+		}
+		currInt := dataValue.(model.Tier0Interface)
+		isT0, gwID, _, _ := parseGatewayInterfacePolicyPath(*currInt.Path)
+
+		if isT0 && gwID == *t0GwList[0].Id {
+			isOp = true
+			if currInt.Path != nil {
+				err := d.Set("path", *currInt.Path)
+				if err != nil {
+					return fmt.Errorf("Error while setting interface path : %v", err)
+				}
+			}
+			if isPolicyGlobalManager(m) {
```

Comment:
Unlike T0, maybe for gateway interfaces even on GM we might be able to set the edge cluster path.
Can we verify this? I don't think a given interface will span multiple LM instances

Reply:
@annakhm can you please comment on this?